protocol/validation/validation.go

```diff
@@ -376,15 +376,21 @@ func checkValidDest(vs *validationState, vd *bc.ValueDestination) error {
 	return nil
 }
 
+// ValidateBlockSig runs the consensus program from prev on b.
+func ValidateBlockSig(b, prev *bc.BlockEntries) error {
+	vmContext := NewBlockVMContext(b, prev.Body.NextConsensusProgram, b.Witness.Arguments)
```

Comment:
Y'know, haha, no bulletproofing would be needed if we did the thing here that we were talking about in [in this thread](https://github.com/chain/chain/pull/788#discussion_r108822154): have callers pass not `prev` but only its `NextConsensusProgram`, which is all this function wants.